sdk/tests/conformance/renderbuffers/depth-renderbuffer-initialization.html

```diff
@@ -0,0 +1,145 @@
+        runTest(gl, {alloc1: {w: c.width, h: c.height}, alloc2: null});
+        runTest(gl, {alloc1: null, alloc2: {w: c.width, h: c.height}});
+        // Tests for initially allocating at the wrong size.
+        runTest(gl, {alloc1: {w: 5, h: 5}, alloc2: {w: c.width, h: c.height}});
+    }
+
+    // Testing buffer clearing won't change the clear values.
+    var clearColor = gl.getParameter(gl.COLOR_CLEAR_VALUE);
+    shouldBe("clearColor", "[0, 1, 0, 1]");
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, 'should be no errors');
+}
+
+function runTest(gl, params) {
+    debug("Test for depth buffer: " + JSON.stringify(params));
+    let final = params.alloc2 ? params.alloc2 : params.alloc1;
+    gl.viewport(0, 0, final.w, final.h);
+    wtu.checkCanvasRect(gl, 0, 0, final.w, final.h,
+                        [0, 0, 0, 0],
+                        "internal buffers have been initialized to 0");
+
+    gl.disable(gl.DEPTH_TEST);
+
+    // fill the back buffer so we know that reading below happens from
+    // the renderbuffer.
+    gl.clearDepth(0);
+    gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+
+    // Set up (color+depth) test buffer.
+    var fbo = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+    var buffer = gl.createRenderbuffer();
+    var depthBuffer = gl.createRenderbuffer();
+
+    if (params.alloc1) {
+        gl.bindRenderbuffer(gl.RENDERBUFFER, buffer);
+        allocStorage(gl.RGBA4, params.alloc1.w, params.alloc1.h);
+        gl.bindRenderbuffer(gl.RENDERBUFFER, depthBuffer);
+        allocStorage(gl.DEPTH_COMPONENT16, params.alloc1.w, params.alloc1.h);
+    }
+    gl.bindRenderbuffer(gl.RENDERBUFFER, buffer);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, buffer);
+    gl.bindRenderbuffer(gl.RENDERBUFFER, depthBuffer);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.DEPTH_ATTACHMENT, gl.RENDERBUFFER, depthBuffer);
+    if (params.alloc2) {
+        if (params.alloc1) {
+            // Clear the FBO in order to make sure framebufferRenderbuffer is
+            // committed.
+            gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+        }
+        gl.bindRenderbuffer(gl.RENDERBUFFER, buffer);
+        allocStorage(gl.RGBA4, params.alloc2.w, params.alloc2.h);
+        gl.bindRenderbuffer(gl.RENDERBUFFER, depthBuffer);
+        allocStorage(gl.DEPTH_COMPONENT16, params.alloc2.w, params.alloc2.h);
+    }
+
+    function allocStorage(format, width, height) {
+        gl.renderbufferStorage(gl.RENDERBUFFER, format, width, height);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+            "should be no error after renderbufferStorage.");
+    }
+
+    shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)",
+             "gl.FRAMEBUFFER_COMPLETE");
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, 'should be no errors');
+
+    // Draw a blue quad (at z = 0.0, so depth = 0.5) (should pass the depth test)
+    gl.depthFunc(gl.LESS);
```

Comment:
The current depth should be initialized to 0? so depth == 0.5 with gl.LESS should not pass the depth test actually. Or am I missing something here?

Reply:
Please use 'depth' to refer to the [0-1] depth range, and 'clip z' to refer to [-1,1] clip space coordinates.

Reply:
When depth is uninitialized, it defaults to 1.0. So the depth test should pass.